core/corehttp/gateway_handler.go

```diff
@@ -133,6 +133,7 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	}
 
 	w.Header().Set("X-IPFS-Path", urlPath)
+	w.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
@slang800 do you lose anything if we allow you to specify gateway headers thus:

```
ipfs config Gateway.Headers.Access-Control-Allow-Origin "*"
# or 
ipfs config Gateway.Headers.Access-Control-Allow-Origin "<origin>"
# or
ipfs config Gateway.Headers.Access-Control-Allow-Origin "mysite.com"
```
?